31 error handling in case ntp server is unreachable (#172)
IHE/DEV.SDPi: asciidoc/volume1/use-cases/tf1-ch-c-use-case-stad.adoc

NOTE: A <<term_manufacturer>> may decide to limit user notification of technical issues to certain user groups (e.g., biomed).

====== Safety, Effectiveness & Security Considerations and Requirements

.R1530
[sdpi_requirement#r1530,sdpi_req_level=shall]
****
If a <<vol1_spec_sdpi_p_actor_somds_participant>> is operational and loses connection to the <<acronym_ts_service>>, it shall use its internal clock.

.Notes
[%collapsible]
====
NOTE: It is likely that a <<vol1_spec_sdpi_p_actor_somds_participant>> needs multiple attempts to connect to a TS service a few times during the day. The system needs to be stable against these kind of short term interruptions.
====
****

.R1531
[sdpi_requirement#r1531,sdpi_req_level=shall]
****

> If a <<vol1_spec_sdpi_p_actor_somds_provider>> cannot reach the <<acronym_ts_service>> and can no longer guarantee a 50ms accuracy, it shall provide ClockState/@Accuracy.

Author: 50 ms is what we wrote in the ntp server configuration as this is sufficient for even the most demanding use cases. The idea was that consumers can rely on this accuracy, if everything works as expected. As soon as that changes, they need to decide for themselves and consequently need all information required for a decision.

.Notes
[%collapsible]
====
NOTE: This allows the <<vol1_spec_sdpi_p_actor_somds_consumer>> to decide if timestamps are still accurate enough for its specific use case.
====
****
*REVIEWER QUESTION*:Is R1531 too challenging for legacy devices?

.R1532
[sdpi_requirement#r1532,sdpi_req_level=shall]
****

> The <<term_manufacturer>> of a <<vol1_spec_sdpi_p_actor_somds_consumer>> shall consider the risk of providing the <<vol1_spec_sdpi_p_actor_somds_consumer>>'s <<term_system_function_contribution>>, if the accuracy of the device internal clock decreases due to an unreachable <<acronym_ts_service>>.

Reviewer: ```suggestion
The <<term_manufacturer>> of a <<vol1_spec_sdpi_p_actor_somds_consumer>> shall consider the risk of providing the <<vol1_spec_sdpi_p_actor_somds_consumer>>'s <<term_system_function_contribution>> if the accuracy of the device internal clock decreases due to an unreachable <<acronym_ts_service>>.
```

****

.R1533
[sdpi_requirement#r1533,sdpi_req_level=shall]
****

> The <<term_manufacturer>> of a <<vol1_spec_sdpi_p_actor_somds_consumer>> shall consider the risk of providing the <<vol1_spec_sdpi_p_actor_somds_consumer>>'s <<term_system_function_contribution>>, if the accuracy of the <<vol1_spec_sdpi_p_actor_somds_provider>>'s timestamps decreases.

Reviewer: ```suggestion
The <<term_manufacturer>> of a <<vol1_spec_sdpi_p_actor_somds_consumer>> shall consider the risk of providing the <<vol1_spec_sdpi_p_actor_somds_consumer>>'s <<term_system_function_contribution>> if the accuracy of the <<vol1_spec_sdpi_p_actor_somds_provider>>'s clock decreases.
```

.Notes
[%collapsible]
====

> NOTE: The <<vol1_spec_sdpi_p_actor_somds_consumer>> can assume that the accuracy is at least 50 ms, if no other value is provided in the <<term_medical_data_information_base>> ClockState/@Accuracy.

Reviewer: ```suggestion
NOTE: In accordance with <<r1531>>, a <<vol1_spec_sdpi_p_actor_somds_consumer>> can assume that the accuracy of a <<vol1_spec_sdpi_p_actor_somds_provider>>'s clock is at least 50ms if no other value is provided in the <<term_medical_data_information_base>> pm:ClockState/@Accuracy.
```

NOTE: This goes beyond considering the risk for erroneous timestamps required by the Base Participant Key Purpose Standard, since it forces the <<term_manufacturer>> of a <<vol1_spec_sdpi_p_actor_somds_consumer>> to define a minimum accuracy acceptable for a <<term_system_function_contribution>>.

====
****

*REVIEWER QUESTION*:Do we need a requirement, for notifying the biomed in case the <<acronym_ts_service>> is no longer reachable? Or is the following logging requirement sufficient?

.R1534
[sdpi_requirement#r1534,sdpi_req_level=shall]
****

> The <<vol1_spec_sdpi_p_actor_somds_participant>> shall create a log entry, if the <<acronym_ts_service>> cannot be reached.

Reviewer: ```suggestion
If a <<vol1_spec_sdpi_p_actor_somds_participant>> cannot reach the <<acronym_ts_service>>, the <<vol1_spec_sdpi_p_actor_somds_participant>> shall create a log entry.
```

===== Scenario: <<acronym_stad>> {var_use_case_id}.5 - Devices are operational in the MD LAN network but cannot access the TS Service and clock drift is unacceptable

*Given* The <<vol1_spec_sdpi_p_actor_somds_consumer>> is operational on the <<acronym_md_lan>> network

*And* The <<acronym_ts_service>> is no longer operational or otherwise inaccessible

*When* The clock drift of the device internal clock exceeds an internal threshold

*Or* The timestamps of the received data are no longer accurate enough

*Then* The device will notify the user that time synchronization is no longer functional, which will limit the availability of SDC System Functionality

*And* The device will create a log entry noting inaccurate time synchronization

*And* The device will periodically attempt to reconnect to the <<acronym_md_lan>> and <<acronym_ts_service>>

*And* Based on a <<term_manufacturer>>'s risk management, the device may be disconnected entirely from the <<acronym_md_lan>> network.

> NOTE: It is the <<vol1_spec_sdpi_p_actor_somds_consumer>>'s responsibility to decide if timestamps are still accurate enough to execute its <<term_system_function_contribution>>.

Reviewer: ```suggestion
NOTE: It is the <<vol1_spec_sdpi_p_actor_somds_consumer>>'s responsibility to decide if timestamps are accurate enough to execute its <<term_system_function_contribution>>.
```

====== Safety, Effectiveness & Security Considerations and Requirements

.R1540
[sdpi_requirement#r1540,sdpi_req_level=shall]
****

> If a <<vol1_spec_sdpi_p_actor_somds_consumer>> receives an <<term_medical_data_information_base>> containing a ClockState/@Accuracy that is no longer sufficient for at least one of its <<term_system_function_contribution>>, it shall disable all affected <<term_system_function_contribution>>s.

Reviewer: ```suggestion
If a <<vol1_spec_sdpi_p_actor_somds_consumer>> receives an <<term_medical_data_information_base>> containing a pm:ClockState/@Accuracy that is no longer sufficient for at least one of its <<term_system_function_contribution>>s, the <<vol1_spec_sdpi_p_actor_somds_consumer>> shall disable all affected <<term_system_function_contribution>>s.
```

****

.R1541
[sdpi_requirement#r1541,sdpi_req_level=shall]
****

> If a <<vol1_spec_sdpi_p_actor_somds_consumer>>'s  internal clock is no longer sufficient for at least one of its <<term_system_function_contribution>>, it shall disable all affected <<term_system_function_contribution>>s.

Reviewer: ```suggestion
If a <<vol1_spec_sdpi_p_actor_somds_consumer>>'s internal clock is no longer sufficient for at least one of its <<term_system_function_contribution>>s, the <<vol1_spec_sdpi_p_actor_somds_consumer>> shall disable all affected <<term_system_function_contribution>>s.
```

****

.R1542
[sdpi_requirement#r1542,sdpi_req_level=shall]
****

> If a <<vol1_spec_sdpi_p_actor_somds_consumer>> disables one or more <<term_system_function_contribution>>s it shall notify the user.

Reviewer: ```suggestion
When a <<vol1_spec_sdpi_p_actor_somds_consumer>> disables one or more <<term_system_function_contribution>>s, the <<vol1_spec_sdpi_p_actor_somds_consumer>> shall notify the user.
```

****

.R1543
[sdpi_requirement#r1543,sdpi_req_level=shall]
****

> If a <<vol1_spec_sdpi_p_actor_somds_consumer>> disables one or more <<term_system_function_contribution>>s it shall create a log entry, noting the disabled <<term_system_function_contribution>>s as well as the cause for disabling them.

Reviewer: ```suggestion
If a <<vol1_spec_sdpi_p_actor_somds_consumer>> disables one or more <<term_system_function_contribution>>s, the <<vol1_spec_sdpi_p_actor_somds_consumer>> shall create a log entry, noting the disabled <<term_system_function_contribution>>s as well as the cause for disabling them.
```
